packages/wrangler/src/pages/functions/routes-transformation.ts

```diff
@@ -36,12 +39,61 @@ export function convertRoutesToGlobPatterns(
 	return Array.from(new Set(convertedRoutes));
 }
 
+/**
+ * Converts Functions routes like /foo/:bar to a Routing object that's used
+ * to determine if a request should run in the Functions user-worker.
+ * Also consolidates redundant routes such as [/foo/bar, /foo/:bar] -> /foo/*
+ *
+ * @returns RotuesJSONSpec to be written to _routes.generated.json
+ */
 export function convertRoutesToRoutesJSONSpec(
 	routes: RoutesJSONRouteInput
 ): RoutesJSONSpec {
+	// The initial routes coming in are sorted most-specific to least-specific.
+	// The order doesn't have any affect on the output of this function, but
+	// it should speed up route consolidation with less-specific routes being first.
+	routes.reverse();
```

Comment:
Probably not a big deal, but the sorting array functions in JS mutate in place (that includes the `.reverse()`) function. It'd be prudent for this function to not have any side effects and to `const reversedRoutes = [...routes].reverse()`

Reply:
Good catch! Fixing